pkg/http/server.go

```diff
@@ -381,3 +413,164 @@ func (s *Server) HTTPSURL() string {
 	}
 	return ""
 }
+
+func (s *Server) existCertificate(uDomain string) bool {
+	if _, err := os.Stat("./lets_encrypt/" + uDomain + ".key"); os.IsNotExist(err) {
+		return false
+	}
+	if _, err := os.Stat("./lets_encrypt/" + uDomain + ".crt"); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func (s *Server) getLetsEncrypt(uDomain string) error {
+	letsEncryptURL := utils.LetsEncryptStaging
+	certDirectory := "./lets_encrypt"
+	if s.Production {
+		letsEncryptURL = utils.LetsEncryptProduction
+	}
+	err := os.MkdirAll(certDirectory, 0644)
+	if err != nil {
+		return err
+	}
+	// Generate account key
+	aKey, err := newKey(path.Join(certDirectory, "account.pem"))
+	if err != nil {
+		return err
+	}
+
+	// Create the ACME client
+	c := &acme.Client{
+		Key:          aKey,
+		DirectoryURL: letsEncryptURL,
+	}
+
+	// Begin with registration
+	log.Infof("Attempting registration...")
+	_, err = c.Register(context.TODO(), nil, func(string) bool { return true })
+	if err != nil {
+		return err
+	}
+	log.Infof("Registration succeeded")
+
+	// Attempt authorization
+	auth, err := c.Authorize(context.TODO(), s.Domain)
+	if err != nil {
+		return err
+	}
+	var challenge *acme.Challenge
+	for _, c := range auth.Challenges {
+		if c.Type == "http-01" {
+			challenge = c
```

Comment:
I'm guessing you are interested in the first occurrence of a challenge of type `http-01`, if so you could add `break` here to break out of the loop as soon as you find it.